source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRMediaTarget.mm

```diff
@@ -0,0 +1,195 @@
+//
+//  ACRMediaTarget
+//  ACRMediaTarget.mm
+//
+//  Copyright © 2018 Microsoft. All rights reserved.
+//
+
+#import "ACRMediaTarget.h"
+#import "ACRAVPlayerViewHoldingUIView.h"
+#import "ACOHostConfigPrivate.h"
+
+// tags for easy accessing of subviews
+const int playIconTag = 0x49434F4E;
+const int posterTag = 0x504F5354;
+
+@implementation ACRMediaTarget
+{
+    ACOMediaEvent *_mediaEvent;
+    AVPlayerViewController *_mediaViewController;
+    NSURL *_url;
+    NSString *_mimeType;
+    __weak ACRView *_view;
+    __weak UIView *_containingview;
+    __weak UIView<ACRIContentHoldingView> *_superview;
+    BOOL isInline;
+}
+// dedicated initializer
+- (instancetype)initWithMediaEvent:(ACOMediaEvent *)mediaEvent
+                               url:(NSURL *)url
+                          rootView:(ACRView *)rootView
+                            config:(ACOHostConfig *)config
+                    containingview:(UIView *)containingview
+                         superview:(UIView<ACRIContentHoldingView> *)superview
```

Comment:
I'm having trouble following this. The media target has a root view, a containing view, a superview, and optionally also a parent view controller provided by the media delegate. Does it need all these containers? The superview, for one, seems to be unused.

Reply:
Ben, thanks for the feedback; yes, superview is not used; its use was removed after i made changes to rely on ios api and tag to access the superview. root view provides context of adaptive cards, containing view is the superview that holds poster and media views. parent view controller is removed.

---
In reply to: [207006216](https://github.com/Microsoft/AdaptiveCards/pull/1736#discussion_r207006216) [](ancestors = 207006216)